metrics/multilabelmap.go

```diff
@@ -13,6 +13,13 @@ import (
 	"sync"
 )
 
+// MetricType is a Prometheus metric type. At the moment we only support
+// counter and gauge.
+type MetricType string
+
+const Counter MetricType = "counter"
+const Gauge MetricType = "gauge"
```

Comment:
Any descriptions needed on here? If you don't want to describe them all in one you could put a comment block around `const (...)`:

```suggestion
// Prometheus metric types.
const (
	Counter MetricType = "counter"
	Gauge   MetricType = "gauge"
)
```

Reply:
I wonder if maybe in this case, a better approach would be to add specific constructors, e.g., `NewMultiLabelGaugeMap` / `NewMultiLabelCounterMap`, and then migrate the existing uses to those and unexport the field?

(Not intended to block this review, this just seems to me like a better solution than adding constants if we're not going to even use them everywhere)

Reply:
My general rule for constructors vs exported fields is to use constructors for required fields and fields for optional things. And the Type and Help are both optional.